src/bulk/common.ts

```diff
@@ -289,6 +289,7 @@ export class BulkWriteResult {
     if (index < this.result.writeErrors.length) {
       return this.result.writeErrors[index];
     }
+    return;
```

Comment:
```suggestion
return index < this.result.writeErrors.length ? this.result.writeErrors[index] : undefined
```

Reply:
🧑‍💻  ⛳ ?
I'm okay with the suggestion but I don't think we need to be writing `return undefined;` anywhere, which I know this isn't but it's implicitly similar. Does that make sense?

Reply:
What about `this.result.writeErrors[index]`?

Reply:
For some reason I thought that threw an error if index was out of range.  I guess not - I'd prefer this.

I actually prefer `return undefined` but that's just preference.  I like that it's explicit 🤷 but I don't feel terribly strongly about this

Reply:
return undefined would read weird to me personally, and in general, and especially in holding with TS conception of types, if your function is supposed to return something but you don't have a result, the appropriate default is `null` not undefined

Reply:
I agree, I went with the ternary though to preserve the API. Filed NODE-4068